docs/core/diagnostics/debug-threadpool-starvation.md

```diff
@@ -0,0 +1,312 @@
+---
+title: Debug Threadpool Starvation
+description: A tutorial that walks you through debugging and fixing a threadpool starvation issue on .NET Core
+ms.topic: tutorial
+ms.date: 04/19/2022
+---
+
+# Debug Threadpool Starvation
+
+**This article applies to: ✔️** .NET Core 3.1 and later versions
+
+In this tutorial, you'll learn how to debug a threadpool starvation scenario. Threadpool starvation occurs when the threadpool has no available threads to process new work items and it often causes applications to respond very slowly. Using the provided example [ASP.NET Core web app](/samples/dotnet/samples/diagnostic-scenarios), you can cause threadpool starvation intentionally and learn how to diagnose it.
+
+In this tutorial, you will:
+
+> [!div class="checklist"]
+>
+> - Investigate an app that is responding to requests slowly
+> - Use the dotnet-counters tool to identify threadpool starvation is likely occuring
+> - Use the dotnet-stack tool to determine what work is keeping the threadpool threads busy
+
+## Prerequisites
+
+The tutorial uses:
+
+- [.NET Core 6.0 SDK](https://dotnet.microsoft.com/download/dotnet) to build and run the sample app
+- [Sample web app](/samples/dotnet/samples/diagnostic-scenarios) which will demonstrate threadpool starvation behavior
+- [Bombadier](https://github.com/codesenberg/bombardier/releases) generates load for the sample web app
+- [dotnet-counters](dotnet-counters.md) to observe performance counters
+- [dotnet-stack](dotnet-stack.md) to examine thread stacks
+
+## Running the sample app
+
+1. Download the code for the [sample app](/samples/dotnet/samples/diagnostic-scenarios) and build it using the .NET SDK:
+
+  ```dotnetcli
+  E:\demo\DiagnosticScenarios>dotnet build
+  Microsoft (R) Build Engine version 17.1.1+a02f73656 for .NET
+  Copyright (C) Microsoft Corporation. All rights reserved.
+  
+    Determining projects to restore...
+    All projects are up-to-date for restore.
+    DiagnosticScenarios -> E:\demo\DiagnosticScenarios\bin\Debug\net6.0\DiagnosticScenarios.dll
+  
+  Build succeeded.
+      0 Warning(s)
+      0 Error(s)
+  
+  Time Elapsed 00:00:01.26
+  ```
+
+1. Run the app:
+
+  ```dotnetcli
+  E:\demo\DiagnosticScenarios>bin\Debug\net6.0\DiagnosticScenarios.exe
+  info: Microsoft.Hosting.Lifetime[14]
+        Now listening on: http://localhost:5000
+  info: Microsoft.Hosting.Lifetime[14]
+        Now listening on: https://localhost:5001
+  info: Microsoft.Hosting.Lifetime[0]
+        Application started. Press Ctrl+C to shut down.
+  info: Microsoft.Hosting.Lifetime[0]
+        Hosting environment: Production
+  info: Microsoft.Hosting.Lifetime[0]
+        Content root path: E:\demo\DiagnosticScenarios
+  ```
```

Comment:
```suggestion
    ```dotnetcli
    E:\demo\DiagnosticScenarios>bin\Debug\net6.0\DiagnosticScenarios.exe
    info: Microsoft.Hosting.Lifetime[14]
          Now listening on: http://localhost:5000
    info: Microsoft.Hosting.Lifetime[14]
          Now listening on: https://localhost:5001
    info: Microsoft.Hosting.Lifetime[0]
          Application started. Press Ctrl+C to shut down.
    info: Microsoft.Hosting.Lifetime[0]
          Hosting environment: Production
    info: Microsoft.Hosting.Lifetime[0]
          Content root path: E:\demo\DiagnosticScenarios
    ```
```